invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/FileUploader/FileModification/ModificationModal.js

```diff
@@ -0,0 +1,246 @@
+  ModalActions,
+  ModalContent,
+  ModalHeader,
+  Table,
+  TableBody,
+  TableHeader,
+  TableHeaderCell,
+  TableRow,
+} from "semantic-ui-react";
+import RadioGroup from "./RadioGroup";
+
+export class ModificationModal extends Component {
+  constructor(props) {
+    super(props);
+    this.initState = {
+      loading: false,
+      error: undefined,
+      checklistState: [undefined],
+      checkboxState: [false],
+      messages: [],
+    };
+    this.state = { ...this.initState };
+    this.checklist = [
+      {
+        label: i18next.t("I want to update the files with a new version"),
+        message: i18next.t("Instead, you can make a new version"),
+      },
+    ];
+  }
+
+  handleClose = () => {
+    this.setState({
+      ...this.initState,
+    });
+    const { handleClose } = this.props;
+    handleClose();
+  };
+
+  handleRadioUpdate = (index, value) => {
+    const { checklistState } = this.state;
+    const nextChecklistState = checklistState.map((c, i) => {
+      if (i === index) {
+        return value;
+      } else {
+        return c;
+      }
+    });
+    const filteredChecklist = this.checklist.filter((_, index) => {
+      return nextChecklistState[index];
+    });
+    const newMessages = filteredChecklist.map((x) => x["message"]);
+    this.setState({ checklistState: nextChecklistState, messages: newMessages });
+  };
+
+  handleCheckboxUpdate = (index) => {
+    const { checkboxState } = this.state;
+    const nextCheckboxState = checkboxState.map((c, i) => {
+      if (i === index) {
+        return !checkboxState[i];
+      } else {
+        return c;
+      }
+    });
+    this.setState({ checkboxState: nextCheckboxState });
+  };
+
+  handleSubmit = async (values) => {
+    this.setState({ loading: true });
+    const { record } = this.props;
+    const payload = {
+      reason: values.reason,
+      comment: values.comment,
+    };
+    if ("file_modification" in record.links) {
+      this.cancellableAction = withCancel(
+        http.post(record.links.file_modification, payload)
+      );
+      try {
+        const response = await this.cancellableAction.promise;
+        const data = response.data;
+
+        if (response.status === 200) {
+          window.location.reload();
+        } else if (response.status === 201) {
+          window.location.href = data.links.self_html;
+        }
+      } catch (error) {
+        this.setState({ error: error });
+        console.error(error);
+      } finally {
+        this.setState({ loading: false });
+      }
+    } else {
+      this.setState({
+        error: "Could not submit file modification request",
+        loading: false,
+      });
+    }
+  };
+
+  render() {
+    const { open, fileModification } = this.props;
+    const { loading, error, messages, checklistState, checkboxState } = this.state;
+
+    const formDisabled =
+      checkboxState.some((v) => v === false) ||
+      checklistState.some((x) => x === true || x === undefined);
```

Comment:
it is a little bit ambiguous what is the logic to disable the form. could you add a comment ?

also I think formik should be able to handle this in its state if the checkbox fields are required, why do you manually do the form validation?